fairlearn/grid_search.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""
+This module implements a 1d grid search for binary classification with a single
+binary protected attribute.
+
+This is a simplification of the full exponentiated gradient search
+and produces a sequence of models, each of which corresponds to a
+different accuracy/fairness trade-off. The dimensionality of the search
+increases with the number of possible values for the protected attribute,
+so this technique does not scale beyond a binary protected attribute.
+"""
+
+import copy
+import numpy as np
+
+class BinaryClassificationGridSearch:
+    def __init__(self):
+        # Nothing to do
+        return
+
+    def _generate_p0_p1(self, y):
+        """ Function to compute p0 and p1 for the given
+        set of labels
+        """
+        unique, counts = np.unique(y, return_counts=True)
+
+        if len(unique) != 2:
+            raise RuntimeError("Supplied y not binary")
+
+        if not np.array_equal(unique, [0, 1]):
+            raise RuntimeError("Supplied y labels not 0 or 1")
+
```

Comment:
I think our code should also work when all the labels are 0 or 1, rather than both 0 and 1.

Reply:
There are definitely more checks for `None` and empty arrays to be added.

For this specific comment, if all the labels are 0 or 1, doesn't that give a trivial model? How is this case regarded by users - in other packages, would supplying a dataset with only a single label be considered an error, or would the user still expect to get a model back?

Reply:
Yes, the user always expects a model back. You can print a warning, though, just in case the user isn't aware of it.

Reply:
For right now, I'm removing all these checks, since getting them right is going to be non-trivial